01.12.2025/binaryCompare.c

```diff
@@ -0,0 +1,68 @@
+#include <limits.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof(arr[0]))
+
+/*
+ * Compare two unsigned numbers.
+ * Return 0, if a == b,
+ * Return 1, if a > b,
+ * Return -1, if a < b.
+ */
+int binaryCompare(unsigned int a, unsigned int b)
+{
+    const int totalBits = sizeof(a) * 8;
+    /* Start compate from the biggest byte. */
+    for (int bit = totalBits - 1; bit >= 0; bit--) {
+        bool hasBitA = (a & (1 << bit));
+        bool hasBitB = (b & (1 << bit));
```

Comment:
Кажется, 
```suggestion
        bool hasBitA = ((a >> bit) & 1);
        bool hasBitB = ((b >> bit) & 1);
```
читалось бы проще. 

В текущей реализации пришлось чесать репу о том, почему это правда, в смысле, что при касте в bool всё, что не 0 становится true.